flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcRowDataLookupFunction.java

```diff
@@ -146,10 +148,12 @@ public void eval(Object... keys) {
         if (cache != null) {
             List<RowData> cachedRows = cache.getIfPresent(keyRow);
             if (cachedRows != null) {
-                for (RowData cachedRow : cachedRows) {
-                    collect(cachedRow);
+                if (!cachedRows.isEmpty() || !excludeEmptyQueryResult) {
+                    for (RowData cachedRow : cachedRows) {
```

Comment:
i am afraid that you should not just ignore the empty values, instead you should remove the keyRow from the cache if exlucdeEmptyQueryResult is true, and collect is still needed.

Reply:
Hi @wenlong88 , changes has been done, please review

Reply:
Hi @wenlong88 , pipeline got succeed, can you please review and merge

Reply:
hi, you need to add a test cover the cache changes, refer to JdbcRowDataLookupFunctionTest#testEval, on the other hands, I think it is more clearer if you do not put empty to cache in eval instead of put to cache and then invalidate.

Reply:
cool, got it, making changes

Reply:
Hi @wenlong88, code changes has been done, but for test case, I have a doubt, If you look at JdbcRowDataLookupFunctionTest#testEval, it test for collector, but for my use case, output will never get collected in collector, Also there is no getter for cache for which i can assert, should I put a getter for cache in this class, can you please help me with test case for it

Reply:
Hi @wenlong88 , did you checked?

Reply:
you should verify that the empty value is not in the cache, will not affect other eval. for example, eval for the time with empty result, update the db, eval another time, the second eval could get the latest result .

Reply:
@wenlong88 , But for this, I need getter for `cache` instance in JdbcRowDataLookupFunction, should I add it? I can't test with collector object, correct?

Reply:
@wenlong88  added test cases, please review

Reply:
@wenlong88 can we resolve conversations now? Also @wuchong will get it merge?

Reply:
@wenlong88  anything else is remaining in this?

Reply:
@gaurav726 the pr looks good for me now, but you still need to wait for Committers to join the review.

Reply:
cool, thanks for the update @wenlong88

Reply:
@wuchong can you please review my PR.